File: urbanairship-react-native/android/src/main/java/com/urbanairship/reactnative/events/DeepLinkEvent.java
/* Copyright Urban Airship and Contributors */

package com.urbanairship.reactnative.events;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.WritableMap;
import com.urbanairship.reactnative.Event;

/**
 * Deep link event.
 */
public class DeepLinkEvent implements Event {

    private static final String DEEP_LINK_EVENT = "com.urbanairship.deep_link";
    private static final String DEEP_LINK = "deepLink";
    private final String deepLink;

    /**
     * Default constructor.
     *
     * @param deepLink The deep link.
     */
    public DeepLinkEvent(@NonNull String deepLink) {
        this.deepLink = deepLink;
    }

    @NonNull
    @Override
    public String getName() {
        return DEEP_LINK_EVENT;
    }

    @NonNull
    @Override
    public ReadableMap getBody() {
        WritableMap map = Arguments.createMap();
        map.putString(DEEP_LINK, deepLink);
        return map;
    }

    @Override
    public boolean isForeground() {
        return true;
    }
}

File: urbanairship-react-native/android/src/main/java/com/urbanairship/reactnative/events/DeepLinkEvent.kt
/* Copyright Urban Airship and Contributors */

package com.urbanairship.reactnative.events

import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.ReadableMap
import com.urbanairship.reactnative.Event

/**
 * Deep link event.
 */
class DeepLinkEvent(private val deepLink: String) : Event {

    override val name = DEEP_LINK_EVENT

    override val body: ReadableMap = Arguments.createMap().apply {
        putString(DEEP_LINK, deepLink)
    }

    override val isForeground = true

    companion object {
        private const val DEEP_LINK_EVENT = "com.airship.deep_link"
        private const val DEEP_LINK = "deepLink"
    }
}